jenkins/scripts/dynamic_worker_workflow/fetch_logs.sh

```diff
@@ -0,0 +1,127 @@
+    # only if we triggered the e2e from the capm3 repo it will be cloned under tested_repo
+    # else it is under metal3
+    if [[ -d "/home/metal3ci/tested_repo/_artifacts" ]]; then
+        cp -r /home/metal3ci/tested_repo/_artifacts/ "${LOGS_DIR}/e2e_artifacts"
+    else
+        cp -r /home/metal3ci/metal3/_artifacts/ "${LOGS_DIR}/e2e_artifacts"
+    fi
+fi
+
+fetch_k8s_logs()
+{
+    dir_name="k8s_${1}"
+    kconfig="$2"
+
+    NAMESPACES="$(kubectl --kubeconfig="${kconfig}" get namespace -o jsonpath='{.items[*].metadata.name}' 2> /dev/null)"
+    mkdir -p "${LOGS_DIR}/${dir_name}"
+    for NAMESPACE in ${NAMESPACES}; do
+        mkdir -p "${LOGS_DIR}/${dir_name}/${NAMESPACE}"
+        PODS="$(kubectl --kubeconfig="${kconfig}" get pods -n "${NAMESPACE}" -o jsonpath='{.items[*].metadata.name}' 2> /dev/null)"
+        for POD in ${PODS}; do
+            mkdir -p "${LOGS_DIR}/${dir_name}/${NAMESPACE}/${POD}"
+            kubectl --kubeconfig="${kconfig}" describe pods -n "${NAMESPACE}" "${POD}" \
+                    > "${LOGS_DIR}/${dir_name}/${NAMESPACE}/${POD}/stdout_describe.log" \
+                    2> "${LOGS_DIR}/${dir_name}/${NAMESPACE}/${POD}/stderr_describe.log"
+            CONTAINERS="$(kubectl --kubeconfig="${kconfig}" get pods -n "${NAMESPACE}" "${POD}" -o jsonpath='{.spec.containers[*].name}' 2> /dev/null)"
+            for CONTAINER in ${CONTAINERS}; do
+                mkdir -p "${LOGS_DIR}/${dir_name}/${NAMESPACE}/${POD}/${CONTAINER}"
+                kubectl --kubeconfig="${kconfig}" logs -n "${NAMESPACE}" "${POD}" "${CONTAINER}" \
+                    > "${LOGS_DIR}/${dir_name}/${NAMESPACE}/${POD}/${CONTAINER}/stdout.log" \
+                    2> "${LOGS_DIR}/${dir_name}/${NAMESPACE}/${POD}/${CONTAINER}/stderr.log"
+            done
+            INIT_CONTAINERS="$(kubectl --kubeconfig="${kconfig}" get pods -n "${NAMESPACE}" "${POD}" -o jsonpath='{.spec.initContainers[*].name}' 2> /dev/null)"
+            for CONTAINER in ${INIT_CONTAINERS}; do
+                mkdir -p "${LOGS_DIR}/${dir_name}/${NAMESPACE}/${POD}/init/${CONTAINER}"
+                kubectl --kubeconfig="${kconfig}" logs -n "${NAMESPACE}" "${POD}" "${CONTAINER}" \
+                    > "${LOGS_DIR}/${dir_name}/${NAMESPACE}/${POD}/init/${CONTAINER}/stdout.log" \
+                    2> "${LOGS_DIR}/${dir_name}/${NAMESPACE}/${POD}/init/${CONTAINER}/stderr.log"
+            done
+        done
+    done
+}
+
+# Fetch k8s logs
+fetch_k8s_logs "management_cluster" "/home/metal3ci/.kube/config"
+
+# Fetch Ironic containers logs before pivoting to the target cluster, if they exist
+if [ -d /tmp/"${CONTAINER_RUNTIME}" ] && [ "$(ls /tmp/"${CONTAINER_RUNTIME}"/)" ]; then
+    CONTAINER_LOGS_DIR="${LOGS_DIR}/${CONTAINER_RUNTIME}/before_pivoting"
+    mkdir -p "${CONTAINER_LOGS_DIR}"
+    cp -r /tmp/"${CONTAINER_RUNTIME}"/* "${CONTAINER_LOGS_DIR}"
+fi
+
+# Fetch k8s logs form target cluster
+target_config=$(sudo find /tmp/ -type f -name "kubeconfig*")
```

Comment:
Is this file cleaned up between runs? Otherwise we could end up with multiple kubeconfigs!

Reply:
We are using dynamic vm only for one test, then it is deleted.

Reply:
Ok good!